crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -347,8 +365,11 @@ fn run_flycheck(state: &mut GlobalState, vfs_path: VfsPath) -> bool {
                     if id == flycheck.id() {
                         updated = true;
                         match package.filter(|_| !world.config.flycheck_workspace()) {
-                            Some(package) => flycheck.restart_for_package(package),
-                            None => flycheck.restart_workspace(saved_file.clone()),
+                            Some(package) => {
+                                flycheck.restart_for_package(package, target_is_bin.clone())
```

Comment:
We could break here if the target is a binary and avoid cloning.

Reply:
The clone is fine, this isn't really performance critical code. I'd rather we don't break here as it just makes this more complicated than necessary